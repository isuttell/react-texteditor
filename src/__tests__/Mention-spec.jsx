/**
 * Test Mention component
 */

import React from 'react';
import TestUtils from 'react-dom/test-utils';
import {mount} from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Mention', () => {
  const Mention = require('../Mention/Mention').default;

  it('should render without error', () => {
    let element = React.createElement(Mention, {}, 'test');

    expect(() => TestUtils.renderIntoDocument(element)).not.toThrow();
  });

  it('should match snapshot', () => {
    const wrapper = mount(
      <Mention
        title='Morty'
        className='real-classy-stuff'
      >
        @morty
      </Mention>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
