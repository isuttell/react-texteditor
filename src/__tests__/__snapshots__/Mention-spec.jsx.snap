// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mention should match snapshot 1`] = `
<Mention
  className="real-classy-stuff"
  title="Morty"
>
  <span
    className="real-classy-stuff"
    title="Morty"
  >
    @morty
  </span>
</Mention>
`;
